sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/MiscTests.cs

```diff
@@ -63,6 +63,33 @@ public async Task TokenIssuanceStartObjectModelTest()
             Assert.True(TestHelper.DoesPayloadMatch(Payload.TokenIssuanceStart.ExpectedPayload, httpResponseMessage.Content.ReadAsStringAsync().Result));
         }
 
+        /// <summary>Test the request object to verify the correct HttpStatusCode is respond</summary>
+        [Test]
+        [TestCase(RequestStatusType.Successful, HttpStatusCode.OK)]
+        [TestCase(RequestStatusType.Failed, HttpStatusCode.BadRequest)]
+        [TestCase(RequestStatusType.ValidationError, HttpStatusCode.BadRequest)]
+        [TestCase(RequestStatusType.TokenInvalid, HttpStatusCode.Unauthorized)]
+        public async Task TokenIssuanceStartRequestValidationTest(RequestStatusType requestStatusType, HttpStatusCode httpStatusCode)
```

Comment:
Do we already have a test for response validation to check for internal server error?

Reply:
We do have coverage to make sure response validations throw internal server error. In the TokenIssuanceStartActionTest test in MiscTests.cs. We test all actions and return InternalServerError for bad actions

Reply:
Interesting those weren't failing before.

Reply:
NIT: Next time either of us touch this file, let's change the name of this class.

Reply:
They weren't failing because it was always returning internal server error. It was the request validation logic that threw 500s when they should have been 400s. There was no coverage for this scenario before.